torch/csrc/autograd/autograd_meta.cpp

```diff
@@ -102,6 +102,8 @@ namespace {
 // This function is will ensure that the fw_grad_ is properly a view of the base for inplace ops on
 // Tensors that do not have forward grad originally.
 void AutogradMeta::set_fw_grad(const at::TensorBase& new_grad_base, const at::TensorBase& self_base, uint64_t level, bool is_inplace_op) {
+  TORCH_CHECK(!new_grad_base._fw_grad(level).defined(), "Setting a forward grad that "
```

Comment:
Maybe this should be internal assert?

Reply:
No, the user could try to do `make_dual(foo, make_dual(foo, foo))`. We should have a test for that btw.